pkgs/servers/roon-server/default.nix

```diff
@@ -0,0 +1,64 @@
+{ alsaLib, alsaUtils, cifs-utils, fetchurl, ffmpeg, libav, mono, stdenv }:
+
+stdenv.mkDerivation rec {
+  name = "roon-server";
+  version = "100600401";
+
+  src = fetchurl {
+    url = "http://download.roonlabs.com/updates/stable/RoonServer_linuxx64_${version}.tar.bz2";
+    sha256 = "121mmdh35q4bpgsqhcps6a6q1f4ld9v4hq9gp181bf2n779pk8sh";
+  };
+
+  installPhase = ''
+    runHook preInstall
+
+    # Check script
+    sed -i '3i PATH=$PATH:${lib.makeBinPath [ cifs-utils ffmpeg libav ]}' check.sh
+    sed -i '/check_ulimit$/d' check.sh
+
+    # Start script
+    sed -i '3i PATH=$PATH:${lib.makeBinPath [ alsaUtils cifs-utils ffmpeg libav ]}' start.sh
+
+    # Debug logging
+    sed -i '/--debug--gc=sgen --server/exec "$HARDLINK" --debug --gc=sgen --server "$SCRIPT.exe" "$@" -storagetrace -watchertrace' Appliance/RoonAppliance
+
+    # Binaries
+    sed -i '/# boost ulimit/,+2 d' Appliance/RAATServer
+    sed -i '/# boost ulimit/,+2 d' Appliance/RoonAppliance
+    sed -i '/# boost ulimit/,+2 d' Server/RoonServer
+    sed -i '/ln -sf/ d' Appliance/RAATServer
+    sed -i '/ln -sf/ d' Appliance/RoonAppliance
+    sed -i '/ln -sf/ d' Server/RoonServer
+    mkdir -p $out/opt
+    mv * $out/opt
+    ln -sf $out/opt/RoonMono/bin/mono-sgen $out/opt/RoonMono/bin/RoonServer
+    ln -sf $out/opt/RoonMono/bin/mono-sgen $out/opt/RoonMono/bin/RoonAppliance
+    ln -sf $out/opt/RoonMono/bin/mono-sgen $out/opt/RoonMono/bin/RAATServer
+
+    runHook postInstall
+  '';
+
+  preFixup = ''
```

Comment:
Does it work with `nativeBuildInputs = [ autoPatchelfHook ];` instead?

Reply:
It does not:
```
automatically fixing dependencies for ELF files
searching for dependencies of /nix/store/vfzdfsc4i8haj3kfx3vxaw1sgx18rxw0-roon-server/opt/RoonMono/lib/libMonoSupportW.so
searching for dependencies of /nix/store/vfzdfsc4i8haj3kfx3vxaw1sgx18rxw0-roon-server/opt/RoonMono/lib/libmono-btls-shared.so
searching for dependencies of /nix/store/vfzdfsc4i8haj3kfx3vxaw1sgx18rxw0-roon-server/opt/RoonMono/lib/libMonoPosixHelper.so
searching for dependencies of /nix/store/vfzdfsc4i8haj3kfx3vxaw1sgx18rxw0-roon-server/opt/RoonMono/bin/mono-sgen
searching for dependencies of /nix/store/vfzdfsc4i8haj3kfx3vxaw1sgx18rxw0-roon-server/opt/Appliance/libroonsearch.so
searching for dependencies of /nix/store/vfzdfsc4i8haj3kfx3vxaw1sgx18rxw0-roon-server/opt/Appliance/processreaper
searching for dependencies of /nix/store/vfzdfsc4i8haj3kfx3vxaw1sgx18rxw0-roon-server/opt/Appliance/libroonmedia.so
searching for dependencies of /nix/store/vfzdfsc4i8haj3kfx3vxaw1sgx18rxw0-roon-server/opt/Appliance/check_alsa
  libasound.so.2 -> not found!
```

Reply:
You need to add the libraries it can't find as `buildInput`s. In this case for `libasound` you need the package `alsaLib`.

Reply:
Now it builds but it doesn't run.
```
Jun 25 14:23:44 fractal start.sh[5674]: Stacktrace:
Jun 25 14:23:44 fractal start.sh[5674]:   at <unknown> <0xffffffff>
Jun 25 14:23:44 fractal start.sh[5674]:   at (wrapper managed-to-native) Sooloos.RAAT.RAAT__manager_new (intptr,string,string,string,string,string,string,string,string,bool,Sooloos.RC/RC__LogEntryCallback,intptr,intptr&) [0x0008c] in <8b803de83c154b62b67879ab6e7cc8cf>:0
Jun 25 14:23:44 fractal start.sh[5674]:   at Sooloos.RAATServerApp.ev_go () [0x001d4] in <8b803de83c154b62b67879ab6e7cc8cf>:0
Jun 25 14:23:44 fractal start.sh[5674]:   at System.Threading.ThreadHelper.ThreadStart_Context (object) [0x0001f] in <370a0c27f4b74d1a81431037df6d75bf>:0
Jun 25 14:23:44 fractal start.sh[5674]:   at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext,System.Threading.ContextCallback,object,bool) [0x00073] in <370a0c27f4b74d1a81431037df6d75bf>:0
Jun 25 14:23:44 fractal start.sh[5674]:   at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext,System.Threading.ContextCallback,object,bool) [0x00004] in <370a0c27f4b74d1a81431037df6d75bf>:0
Jun 25 14:23:44 fractal start.sh[5674]:   at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext,System.Threading.ContextCallback,object) [0x0002f] in <370a0c27f4b74d1a81431037df6d75bf>:0
Jun 25 14:23:44 fractal start.sh[5674]:   at System.Threading.ThreadHelper.ThreadStart () [0x00014] in <370a0c27f4b74d1a81431037df6d75bf>:0
Jun 25 14:23:44 fractal start.sh[5674]:   at (wrapper runtime-invoke) object.runtime_invoke_void__this__ (object,intptr,intptr,intptr) [0x0004f] in <370a0c27f4b74d1a81431037df6d75bf>:0
Jun 25 14:23:44 fractal start.sh[5674]: /proc/self/maps:
Jun 25 14:23:44 fractal start.sh[5674]: 003ff000-00400000 rw-p 00000000 fe:00 15598674                           /nix/store/4j6kw1r8rhmjkcslw3f34im1xxiw7d77-roon-server/opt/RoonMono/bin/mono-sgen
Jun 25 14:23:44 fractal start.sh[5674]: 00400000-0096e000 r-xp 00001000 fe:00 15598674                           /nix/store/4j6kw1r8rhmjkcslw3f34im1xxiw7d77-roon-server/opt/RoonMono/bin/mono-sgen
Jun 25 14:23:44 fractal start.sh[5674]: 00b6d000-00b71000 rw-p 0056e000 fe:00 15598674                           /nix/store/4j6kw1r8rhmjkcslw3f34im1xxiw7d77-roon-server/opt/RoonMono/bin/mono-sgen
Jun 25 14:23:44 fractal start.sh[5674]: 00b71000-00bc8000 rw-p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 01b41000-02007000 rw-p 00000000 00:00 0                                  [heap]
Jun 25 14:23:44 fractal start.sh[5674]: 40dee000-40ea3000 rwxp 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 41dd3000-41de3000 rwxp 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7effcc000000-7effcc021000 rw-p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7effcc021000-7effd0000000 ---p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7effd289a000-7effd289b000 ---p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7effd289b000-7effd309b000 rw-p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7effd4000000-7effd4021000 rw-p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7effd4021000-7effd8000000 ---p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7effdb09b000-7effdb09c000 ---p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7effdb09c000-7effdb89c000 rw-p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7effdb89c000-7effdb995000 r-xp 00000000 fe:00 11273963                   /nix/store/cvs6xv383qadls36r3nszwqmk9a81v0v-alsa-lib-1.1.6/lib/libasound.so.2.0.0
Jun 25 14:23:44 fractal start.sh[5674]: 7effdb995000-7effdbb94000 ---p 000f9000 fe:00 11273963                   /nix/store/cvs6xv383qadls36r3nszwqmk9a81v0v-alsa-lib-1.1.6/lib/libasound.so.2.0.0
Jun 25 14:23:44 fractal start.sh[5674]: 7effdbb94000-7effdbb9c000 r--p 000f8000 fe:00 11273963                   /nix/store/cvs6xv383qadls36r3nszwqmk9a81v0v-alsa-lib-1.1.6/lib/libasound.so.2.0.0
Jun 25 14:23:44 fractal start.sh[5674]: 7effdbb9c000-7effdbb9d000 rw-p 00100000 fe:00 11273963                   /nix/store/cvs6xv383qadls36r3nszwqmk9a81v0v-alsa-lib-1.1.6/lib/libasound.so.2.0.0
Jun 25 14:23:44 fractal start.sh[5674]: 7effdbb9d000-7effdbdab000 r-xp 00000000 fe:00 15598583                   /nix/store/4j6kw1r8rhmjkcslw3f34im1xxiw7d77-roon-server/opt/Appliance/libraatmanager.so
Jun 25 14:23:44 fractal start.sh[5674]: 7effdbdab000-7effdbfab000 ---p 0020e000 fe:00 15598583                   /nix/store/4j6kw1r8rhmjkcslw3f34im1xxiw7d77-roon-server/opt/Appliance/libraatmanager.so
Jun 25 14:23:44 fractal start.sh[5674]: 7effdbfab000-7effdbfbc000 rw-p 0020e000 fe:00 15598583                   /nix/store/4j6kw1r8rhmjkcslw3f34im1xxiw7d77-roon-server/opt/Appliance/libraatmanager.so
Jun 25 14:23:44 fractal start.sh[5674]: 7effdbfbc000-7effdbfbf000 rw-p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7effdbfbf000-7effdc000000 rw-p 0070c000 fe:00 15598583                   /nix/store/4j6kw1r8rhmjkcslw3f34im1xxiw7d77-roon-server/opt/Appliance/libraatmanager.so
Jun 25 14:23:44 fractal start.sh[5674]: 7effdc000000-7effdc0d8000 rw-p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7effdc0d8000-7effe0000000 ---p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7effe0000000-7effe0021000 rw-p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7effe0021000-7effe4000000 ---p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7effe4000000-7effe4021000 rw-p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7effe4021000-7effe8000000 ---p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7effe8082000-7effe808d000 r-xp 00000000 fe:00 11276474                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/libnss_files-2.27.so
Jun 25 14:23:44 fractal start.sh[5674]: 7effe808d000-7effe828c000 ---p 0000b000 fe:00 11276474                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/libnss_files-2.27.so
Jun 25 14:23:44 fractal start.sh[5674]: 7effe828c000-7effe828d000 r--p 0000a000 fe:00 11276474                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/libnss_files-2.27.so
Jun 25 14:23:44 fractal start.sh[5674]: 7effe828d000-7effe828e000 rw-p 0000b000 fe:00 11276474                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/libnss_files-2.27.so
Jun 25 14:23:44 fractal start.sh[5674]: 7effe828e000-7effe8294000 rw-p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7effe8294000-7effe8295000 ---p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7effe8295000-7effe8495000 rw-p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7effe8495000-7effe8496000 ---p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7effe8496000-7effe8696000 rw-p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7effe8696000-7effe86fe000 r-xp 00000000 fe:00 15598767                   /nix/store/4j6kw1r8rhmjkcslw3f34im1xxiw7d77-roon-server/opt/RoonMono/lib/libMonoPosixHelper.so
Jun 25 14:23:44 fractal start.sh[5674]: 7effe86fe000-7effe88fd000 ---p 00068000 fe:00 15598767                   /nix/store/4j6kw1r8rhmjkcslw3f34im1xxiw7d77-roon-server/opt/RoonMono/lib/libMonoPosixHelper.so
Jun 25 14:23:44 fractal start.sh[5674]: 7effe88fd000-7effe88ff000 rw-p 00067000 fe:00 15598767                   /nix/store/4j6kw1r8rhmjkcslw3f34im1xxiw7d77-roon-server/opt/RoonMono/lib/libMonoPosixHelper.so
Jun 25 14:23:44 fractal start.sh[5674]: 7effe88ff000-7effe8a00000 rw-p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7effe8a60000-7effe8d61000 r--p 00000000 fe:00 15598716                   /nix/store/4j6kw1r8rhmjkcslw3f34im1xxiw7d77-roon-server/opt/RoonMono/lib/mono/gac/System.Xml/4.0.0.0__b77a5c561934e089/System.Xml.dll
Jun 25 14:23:44 fractal start.sh[5674]: 7effe8d61000-7effe9000000 r--p 00000000 fe:00 15598707                   /nix/store/4j6kw1r8rhmjkcslw3f34im1xxiw7d77-roon-server/opt/RoonMono/lib/mono/gac/System/4.0.0.0__b77a5c561934e089/System.dll
Jun 25 14:23:44 fractal start.sh[5674]: 7effe9000000-7effeb000000 rw-p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7effeb09a000-7effeb0ce000 r--p 00000000 fe:00 15598713                   /nix/store/4j6kw1r8rhmjkcslw3f34im1xxiw7d77-roon-server/opt/RoonMono/lib/mono/gac/Mono.Posix/4.0.0.0__0738eb9f132ed756/Mono.Posix.dll
Jun 25 14:23:44 fractal start.sh[5674]: 7effeb0ce000-7effeb11a000 r--p 00000000 fe:00 15598710                   /nix/store/4j6kw1r8rhmjkcslw3f34im1xxiw7d77-roon-server/opt/RoonMono/lib/mono/gac/Mono.Security/4.0.0.0__0738eb9f132ed756/Mono.Security.dll
Jun 25 14:23:44 fractal start.sh[5674]: 7effeb11a000-7effeb11b000 ---p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7effeb11b000-7effeb31b000 rw-p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7effeb31b000-7effeb6d2000 r--p 00000000 fe:00 15598756                   /nix/store/4j6kw1r8rhmjkcslw3f34im1xxiw7d77-roon-server/opt/RoonMono/lib/mono/4.5/mscorlib.dll
Jun 25 14:23:44 fractal start.sh[5674]: 7effeb6d2000-7effeb6d3000 ---p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7effeb6d3000-7effebed3000 rw-p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7effebed3000-7efff37ea000 r--p 00000000 fe:00 11534392                   /nix/store/xhx74n52mwajpb3cbb351wzg851pq0yw-glibc-locales-2.27/lib/locale/locale-archive
Jun 25 14:23:44 fractal start.sh[5674]: 7efff37ea000-7efff3994000 r-xp 00000000 fe:00 11276439                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/libc-2.27.so
Jun 25 14:23:44 fractal start.sh[5674]: 7efff3994000-7efff3b94000 ---p 001aa000 fe:00 11276439                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/libc-2.27.so
Jun 25 14:23:44 fractal start.sh[5674]: 7efff3b94000-7efff3b98000 r--p 001aa000 fe:00 11276439                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/libc-2.27.so
Jun 25 14:23:44 fractal start.sh[5674]: 7efff3b98000-7efff3b9a000 rw-p 001ae000 fe:00 11276439                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/libc-2.27.so
Jun 25 14:23:44 fractal start.sh[5674]: 7efff3b9a000-7efff3b9e000 rw-p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7efff3b9e000-7efff3bb4000 r-xp 00000000 fe:00 11276453                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/libgcc_s.so.1
Jun 25 14:23:44 fractal start.sh[5674]: 7efff3bb4000-7efff3db3000 ---p 00016000 fe:00 11276453                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/libgcc_s.so.1
Jun 25 14:23:44 fractal start.sh[5674]: 7efff3db3000-7efff3db4000 rw-p 00015000 fe:00 11276453                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/libgcc_s.so.1
Jun 25 14:23:44 fractal start.sh[5674]: 7efff3db4000-7efff3dcd000 r-xp 00000000 fe:00 11276487                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/libpthread-2.27.so
Jun 25 14:23:44 fractal start.sh[5674]: 7efff3dcd000-7efff3fcd000 ---p 00019000 fe:00 11276487                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/libpthread-2.27.so
Jun 25 14:23:44 fractal start.sh[5674]: 7efff3fcd000-7efff3fce000 r--p 00019000 fe:00 11276487                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/libpthread-2.27.so
Jun 25 14:23:44 fractal start.sh[5674]: 7efff3fce000-7efff3fcf000 rw-p 0001a000 fe:00 11276487                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/libpthread-2.27.so
Jun 25 14:23:44 fractal start.sh[5674]: 7efff3fcf000-7efff3fd3000 rw-p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7efff3fd3000-7efff3fd6000 r-xp 00000000 fe:00 11276449                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/libdl-2.27.so
Jun 25 14:23:44 fractal start.sh[5674]: 7efff3fd6000-7efff41d5000 ---p 00003000 fe:00 11276449                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/libdl-2.27.so
Jun 25 14:23:44 fractal start.sh[5674]: 7efff41d5000-7efff41d6000 r--p 00002000 fe:00 11276449                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/libdl-2.27.so
Jun 25 14:23:44 fractal start.sh[5674]: 7efff41d6000-7efff41d7000 rw-p 00003000 fe:00 11276449                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/libdl-2.27.so
Jun 25 14:23:44 fractal start.sh[5674]: 7efff41d7000-7efff41de000 r-xp 00000000 fe:00 11276494                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/librt-2.27.so
Jun 25 14:23:44 fractal start.sh[5674]: 7efff41de000-7efff43dd000 ---p 00007000 fe:00 11276494                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/librt-2.27.so
Jun 25 14:23:44 fractal start.sh[5674]: 7efff43dd000-7efff43de000 r--p 00006000 fe:00 11276494                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/librt-2.27.so
Jun 25 14:23:44 fractal start.sh[5674]: 7efff43de000-7efff43df000 rw-p 00007000 fe:00 11276494                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/librt-2.27.so
Jun 25 14:23:44 fractal start.sh[5674]: 7efff43df000-7efff4572000 r-xp 00000000 fe:00 11276454                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/libm-2.27.so
Jun 25 14:23:44 fractal start.sh[5674]: 7efff4572000-7efff4772000 ---p 00193000 fe:00 11276454                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/libm-2.27.so
Jun 25 14:23:44 fractal start.sh[5674]: 7efff4772000-7efff4773000 r--p 00193000 fe:00 11276454                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/libm-2.27.so
Jun 25 14:23:44 fractal start.sh[5674]: 7efff4773000-7efff4774000 rw-p 00194000 fe:00 11276454                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/libm-2.27.so
Jun 25 14:23:44 fractal start.sh[5674]: 7efff4774000-7efff4799000 r-xp 00000000 fe:00 11276430                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/ld-2.27.so
Jun 25 14:23:44 fractal start.sh[5674]: 7efff47bd000-7efff47bf000 rw-p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7efff47bf000-7efff47dd000 r--p 00000000 fe:00 15598745                   /nix/store/4j6kw1r8rhmjkcslw3f34im1xxiw7d77-roon-server/opt/RoonMono/lib/mono/gac/System.Configuration/4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
Jun 25 14:23:44 fractal start.sh[5674]: 7efff47dd000-7efff47fa000 r--p 00000000 fe:00 15598637                   /nix/store/4j6kw1r8rhmjkcslw3f34im1xxiw7d77-roon-server/opt/Appliance/Base.dll
Jun 25 14:23:44 fractal start.sh[5674]: 7efff47fa000-7efff4868000 r--p 00000000 fe:00 15598636                   /nix/store/4j6kw1r8rhmjkcslw3f34im1xxiw7d77-roon-server/opt/Appliance/RoonBase.dll
Jun 25 14:23:44 fractal start.sh[5674]: 7efff4868000-7efff486c000 r--p 00000000 fe:00 15598598                   /nix/store/4j6kw1r8rhmjkcslw3f34im1xxiw7d77-roon-server/opt/Appliance/RAATServerApp.dll
Jun 25 14:23:44 fractal start.sh[5674]: 7efff486c000-7efff48ed000 rw-p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7efff48ed000-7efff48ee000 rw-p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7efff48ee000-7efff48f0000 r--p 00000000 fe:00 15598612                   /nix/store/4j6kw1r8rhmjkcslw3f34im1xxiw7d77-roon-server/opt/Appliance/RAATServer.exe
Jun 25 14:23:44 fractal start.sh[5674]: 7efff48f0000-7efff48fc000 rw-p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7efff48fc000-7efff495b000 ---p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7efff495b000-7efff495c000 rw-s 00000000 00:12 5850627                    /dev/shm/mono.5825
Jun 25 14:23:44 fractal start.sh[5674]: 7efff495c000-7efff4991000 r--s 00000000 00:13 721580                     /run/nscd/dbnoZqXM (deleted)
Jun 25 14:23:44 fractal start.sh[5674]: 7efff4991000-7efff4998000 rw-p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7efff4998000-7efff4999000 r--p 00024000 fe:00 11276430                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/ld-2.27.so
Jun 25 14:23:44 fractal start.sh[5674]: 7efff4999000-7efff499a000 rw-p 00025000 fe:00 11276430                   /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/ld-2.27.so
Jun 25 14:23:44 fractal start.sh[5674]: 7efff499a000-7efff499b000 rw-p 00000000 00:00 0
Jun 25 14:23:44 fractal start.sh[5674]: 7ffe21281000-7ffe212a2000 rw-p 00000000 00:00 0                          [stack]
Jun 25 14:23:44 fractal start.sh[5674]: 7ffe213b6000-7ffe213b9000 r--p 00000000 00:00 0                          [vvar]
Jun 25 14:23:44 fractal start.sh[5674]: 7ffe213b9000-7ffe213bb000 r-xp 00000000 00:00 0                          [vdso]
Jun 25 14:23:44 fractal start.sh[5674]: ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]
Jun 25 14:23:44 fractal start.sh[5674]: Native stacktrace:
Jun 25 14:23:44 fractal start.sh[5674]:         /nix/store/4j6kw1r8rhmjkcslw3f34im1xxiw7d77-roon-server/opt/RoonMono/bin/RAATServer() [0x5064e6]
Jun 25 14:23:44 fractal start.sh[5674]:         /nix/store/4j6kw1r8rhmjkcslw3f34im1xxiw7d77-roon-server/opt/RoonMono/bin/RAATServer() [0x422d98]
Jun 25 14:23:44 fractal start.sh[5674]:         /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/libpthread.so.0(+0x11f10) [0x7efff3dc5f10]
Jun 25 14:23:44 fractal start.sh[5674]:         /nix/store/7gx4kiv5m0i7d7qkixq2cwzbr10lvxwc-glibc-2.27/lib/libc.so.6(+0x1514b1) [0x7efff393b4b1]
Jun 25 14:23:44 fractal start.sh[5674]:         /nix/store/cvs6xv383qadls36r3nszwqmk9a81v0v-alsa-lib-1.1.6/lib/libasound.so.2(snd_user_file+0x1b7) [0x7effdb8daee7]
Jun 25 14:23:44 fractal start.sh[5674]:         /nix/store/cvs6xv383qadls36r3nszwqmk9a81v0v-alsa-lib-1.1.6/lib/libasound.so.2(snd_config_hook_load+0x342) [0x7effdb8d4ba2]
Jun 25 14:23:44 fractal start.sh[5674]:         /nix/store/cvs6xv383qadls36r3nszwqmk9a81v0v-alsa-lib-1.1.6/lib/libasound.so.2(+0x3992f) [0x7effdb8d592f]
Jun 25 14:23:44 fractal start.sh[5674]:         /nix/store/cvs6xv383qadls36r3nszwqmk9a81v0v-alsa-lib-1.1.6/lib/libasound.so.2(+0x39dd5) [0x7effdb8d5dd5]
Jun 25 14:23:44 fractal start.sh[5674]:         /nix/store/cvs6xv383qadls36r3nszwqmk9a81v0v-alsa-lib-1.1.6/lib/libasound.so.2(snd_config_update_r+0x25c) [0x7effdb8d60cc]
Jun 25 14:23:44 fractal start.sh[5674]:         /nix/store/cvs6xv383qadls36r3nszwqmk9a81v0v-alsa-lib-1.1.6/lib/libasound.so.2(snd_config_update_ref+0x4a) [0x7effdb8d661a]
Jun 25 14:23:44 fractal start.sh[5674]:         /nix/store/cvs6xv383qadls36r3nszwqmk9a81v0v-alsa-lib-1.1.6/lib/libasound.so.2(snd_ctl_open+0x35) [0x7effdb8e0e05]
Jun 25 14:23:44 fractal start.sh[5674]:         /nix/store/4j6kw1r8rhmjkcslw3f34im1xxiw7d77-roon-server/opt/Appliance/libraatmanager.so(+0xbf2e4) [0x7effdbc5c2e4]
Jun 25 14:23:44 fractal start.sh[5674]:         /nix/store/4j6kw1r8rhmjkcslw3f34im1xxiw7d77-roon-server/opt/Appliance/libraatmanager.so(RAAT__manager_platform_new+0x6d) [0x7effdbc5dddd]
Jun 25 14:23:44 fractal start.sh[5674]:         /nix/store/4j6kw1r8rhmjkcslw3f34im1xxiw7d77-roon-server/opt/Appliance/libraatmanager.so(RAAT__manager_new+0x3d1) [0x7effdbc4c201]
Jun 25 14:23:44 fractal start.sh[5674]:         [0x40e96883]
Jun 25 14:23:44 fractal start.sh[5674]: =================================================================
Jun 25 14:23:44 fractal start.sh[5674]: Got a SIGSEGV while executing native code. This usually indicates
Jun 25 14:23:44 fractal start.sh[5674]: a fatal error in the mono runtime or one of the native libraries
Jun 25 14:23:44 fractal start.sh[5674]: used by your application.
Jun 25 14:23:44 fractal start.sh[5674]: =================================================================
```

Reply:
Does ```ldd EXECUTEABLE``` reveal anything about missing libraries?

Reply:
Nothing immediately jumps out. https://gist.github.com/Steell/024335a7c9ff175e704d1eade46a1c12

Reply:
If the autopatcher doesn't cut it, it's perfectly fine to patchelf it yourself.

Reply:
SGTM, marking this as resolved for now